### PR TITLE
io.github.eminfedar.vaktisalah-gtk-rs Application submission

### PR DESCRIPTION
**Pull request by eminfedar**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.eminfedar.vaktisalah-gtk.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "sdk": "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "vaktisalah-gtk",
+    "finish-args": [
+        "--share=network",
+        "--socket=pulseaudio",
+        "--share=ipc",
+        "--socket=x11",
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:14]: if it really supports wayland then this should be `fallback-x11`

File: com.eminfedar.vaktisalah-gtk.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            "config-opts" : [
+                "-Dvapi=false",
+                "-Dexamples=false",
+                "-Dtests=false"
+            ],
+            "sources": [
+                {
+                    "type":"git",
+                    "url":"https://gitlab.gnome.org/GNOME/libadwaita.git",
+                    "tag": "1.3.2"
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:55]: this version of libadwaita is already in the runtime.

File: com.eminfedar.vaktisalah-gtk.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                }
+            ]
+        },
+        {
+            "name": "blueprint-compiler",
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:60]: you need a cleanup rule to remove this build-only package completely

File: com.eminfedar.vaktisalah-gtk.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/eminfedar/vaktisalah-gtk-rs",
+                    "tag": "v1.0.0"
+                  }
+            ],
+            "build-options": {
+                "append-path" : "/usr/lib/sdk/rust-stable/bin",
+                "build-args": [
+                    "--share=network"
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:83]: build args is not allowed. the build must be offline.

File: com.eminfedar.vaktisalah-gtk.json [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                ],
+                "env" : {
+                    "CARGO_HOME" : "/run/build/vaktisalah-gtk/.cargo",
+                    "LOCALIZATION_DIR": "../localization/"
+                }
+            },
+            "build-commands": [
+                "cargo build --release --target x86_64-unknown-linux-gnu",
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:91]: so what happen when the flatpak is built on aarch64?

File: com.eminfedar.vaktisalah-gtk.json [hunk 6]
```diff
@@ -0,0 +1,101 @@
+{
+    "id": "com.eminfedar.vaktisalah-gtk",
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:2]: there is nothing found at `eminfedar.com`.

Comment (eminfedar) [reply com.eminfedar.vaktisalah-gtk.json:2]: Hmm, I think I can change it to "io.github.eminfedar.vaktisalah-gtk" but it still will be a non existing website. This program only needs a github repository, not a separate website.

File: com.eminfedar.vaktisalah-gtk.json (hunk 5 above)
Comment (hfiguiere) [reply com.eminfedar.vaktisalah-gtk.json:91]: why do you specify a target in the first place?

Comment (eminfedar) [reply com.eminfedar.vaktisalah-gtk.json:75]: To copy the result of the compilation to `/app/bin` as I see in examples:
```
                         |
                         V
"install -Dm755 ./target/x86_64-unknown-linux-gnu/release/vaktisalah-gtk-rs -t /app/bin/"
```

Comment (hfiguiere) [reply com.eminfedar.vaktisalah-gtk.json:75]: so you don't need.

File: com.eminfedar.vaktisalah-gtk.json (hunk 6 above)
Comment (barthalion) [reply com.eminfedar.vaktisalah-gtk.json]: `io.github.eminfedar.vaktisalah-gtk` would be preferred even if there's no website at that address.

File: com.eminfedar.vaktisalah-gtk.json (hunk 3 above)
Comment (barthalion) [reply com.eminfedar.vaktisalah-gtk.json:47]: `"cleanup": ["*"]`

File: com.eminfedar.vaktisalah-gtk.json (hunk 6 above)
Comment (barthalion) [reply com.eminfedar.vaktisalah-gtk.json]: Never mind, just noticed it's already done.

File: com.eminfedar.vaktisalah-gtk.json (hunk 1 above)
Comment (barthalion) [reply com.eminfedar.vaktisalah-gtk.json:14]: Still needs removing.

File: com.eminfedar.vaktisalah-gtk.json
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:1]: this file's name should match the app-id

File: com.eminfedar.vaktisalah-gtk.json [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            },
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml",
+                "cargo --offline build --release",
+                "install -Dm755 ./target/release/vaktisalah-gtk-rs -t /app/bin/"
+            ],
+            "post-install": [
+                "install -Dm644 export/com.eminfedar.vaktisalah-gtk-rs.svg /app/share/icons/hicolor/scalable/apps/com.eminfedar.vaktisalah-gtk-rs.svg",
+                "install -Dm644 export/com.eminfedar.vaktisalah-gtk-rs.desktop /app/share/applications/com.eminfedar.vaktisalah-gtk-rs.desktop",
+                "install -Dm644 export/com.eminfedar.vaktisalah-gtk-rs.appdata.xml /app/share/metainfo/com.eminfedar.vaktisalah-gtk-rs.appdata.xml"
```
Comment (hfiguiere) [inline com.eminfedar.vaktisalah-gtk.json:82]: these files are installed with the wrong name, like `rename-*` is the best course. 

The appstream file likely need to have the id patched too.

File: com.eminfedar.vaktisalah-gtk.json [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        },
+        {
+            "name": "sassc",
+            "buildsystem":"meson",
+            "builddir": true,
+            "sources": [
+                {
+                    "type":"git",
+                    "url":"https://github.com/lazka/sassc.git",
+                    "branch": "meson",
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.eminfedar.vaktisalah-gtk.json:41]:
```suggestion

```

[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.eminfedar.vaktisalah-gtk.json:41]:
```suggestion
```

**Comment by barthalion** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
> this version of libadwaita is already in the runtime.


Build failed with no libadwaita error:
```
error: no matching package named `libadwaita` found
```
Should I add it again?

**Comment by hfiguiere** [comment]
you are missing the cargo sources:

see https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo

**Comment by eminfedar** [comment]
Can I run the build bot?

**Comment by eminfedar** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
> > this version of libadwaita is already in the runtime.
> 
> Build failed with no libadwaita error:
> 
> ```
> error: no matching package named `libadwaita` found
> ```
> 
> Should I add it again?

Added cargo sources but still the same error.

File: io.github.eminfedar.vaktisalah-gtk-rs.json [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                    "type": "git",
+                    "url": "https://github.com/eminfedar/vaktisalah-gtk-rs",
+                    "tag": "v1.0.0"
+                },
+                "cargo-sources.json"
+            ],
+            "build-options": {
+                "append-path" : "/usr/lib/sdk/rust-stable/bin",
+                "env" : {
+                    "CARGO_HOME" : "/run/build/vaktisalah-gtk-rs/.cargo",
```
Comment (hfiguiere) [inline io.github.eminfedar.vaktisalah-gtk-rs.json:70]:
```suggestion
                    "CARGO_HOME" : "/run/build/vaktisalah-gtk-rs/cargo",
```

**Comment by eminfedar** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
An interesting AppStream .xml file error on build, I don't know why.

Is it about changing the name from `io.github.eminfedar.vaktisalah-gtk-rs.appstream.xml` -> `io.github.eminfedar.vaktisalah-gtk-rs.xml`? Should the file's extension be like `.appstream.xml`?

File: io.github.eminfedar.vaktisalah-gtk-rs.json [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+            },
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml",
+                "cargo --offline build --release",
+                "install -Dm755 ./target/release/vaktisalah-gtk-rs -t /app/bin/"
+            ],
+            "post-install": [
+                "install -Dm644 export/io.github.eminfedar.vaktisalah-gtk-rs.svg /app/share/icons/hicolor/scalable/apps/io.github.eminfedar.vaktisalah-gtk-rs.svg",
+                "install -Dm644 export/io.github.eminfedar.vaktisalah-gtk-rs.desktop /app/share/applications/io.github.eminfedar.vaktisalah-gtk-rs.desktop",
+                "install -Dm644 export/io.github.eminfedar.vaktisalah-gtk-rs.xml /app/share/metainfo/io.github.eminfedar.vaktisalah-gtk-rs.xml"
```
Comment (hfiguiere) [inline io.github.eminfedar.vaktisalah-gtk-rs.json:82]:
```suggestion
                "install -Dm644 export/io.github.eminfedar.vaktisalah-gtk-rs.xml /app/share/metainfo/io.github.eminfedar.vaktisalah-gtk-rs.metainfo.xml"
```

Ideally the name upstream could be changed as it is incorrect.

File: io.github.eminfedar.vaktisalah-gtk-rs.json [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+            "sources": [
+                {
+                    "type":"git",
+                    "url":"https://github.com/lazka/libsass.git",
+                    "commit": "aac79dccd3c8f7e8f22125f87a119f3b1ee9d487"
+                }
+            ]
+        },
+        {
+            "name": "sassc",
```
Comment (hfiguiere) [inline io.github.eminfedar.vaktisalah-gtk-rs.json:33]: isn't this also only for blueprint?

File: io.github.eminfedar.vaktisalah-gtk-rs.json (hunk 10 above)
Comment (hfiguiere) [reply io.github.eminfedar.vaktisalah-gtk-rs.json:82]: and the original was fine.

File: io.github.eminfedar.vaktisalah-gtk-rs.json (hunk 11 above)
Comment (eminfedar) [reply io.github.eminfedar.vaktisalah-gtk-rs.json:33]: Do you mean I should remove libsass & sassc and only blueprint & vaktisalah itself should remain?

Comment (hfiguiere) [reply io.github.eminfedar.vaktisalah-gtk-rs.json:33]: if it is only needed for blueprint it should have a similar cleanup rule too.

**Comment by eminfedar** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

File: io.github.eminfedar.vaktisalah-gtk-rs.json (hunk 11 above)
Comment (hfiguiere) [reply io.github.eminfedar.vaktisalah-gtk-rs.json:33]: so it's not even needed.

**Comment by barthalion** [comment]
bot, build io.github.eminfedar.vaktisalah-gtk-rs

**Comment by eminfedar** [comment]
@barthalion When I installed this application on flathub, it has dependencies of Nvidia 530 flatpak platform and org.gtk.Gtk3theme.adw-gtk3 package. I didn't set these dependencies. Are they automatically generated from my computer? Because I compiled this program on nvidia drivers and adw-gtk3 theme as my gtk theme. If so, I should recompile it without these selections.

Thanks for your answers.

**Comment by barthalion** [comment]
Yes, they are automatically installed because of your computer configuration.

**Comment by eminfedar** [comment]
@barthalion Thanks for the answer but there is a confusion on my side.

Flatpak's [website](https://docs.flatpak.org/en/latest/building-introduction.html#software-development-kits-sdks) mentions this:

> Instead of being built using the host environment, Flatpak applications are built inside a separate environment, called an SDK.

How can I make my flatpak build separated from my environment?

**Comment by barthalion** [comment]
What makes you think it's not separated? There's no need to recompile anything, it's a feature. When Flatpak detects user is using Nvidia drivers, it installs Nvidia drivers required by the runtime; same about the theme for better desktop integration.

**Pull request closed without merging**